jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -455,6 +455,10 @@ private void extractQueryParameters()
     /* ------------------------------------------------------------ */
     private void extractContentParameters()
     {
+        // Content cannot be encoded
+        if (_metaData!=null && getHttpFields().contains(HttpHeader.CONTENT_ENCODING))
+            throw new BadMessageException(HttpStatus.NOT_IMPLEMENTED_501,"Unsupported Content-Encoding");
```

Comment:
It seems overeager to fail the request before the content type was inspected. As is, requests with content types for which extraction of content parameters isn't in order are failed, too.

Reply:
If a `Content-Encoding` is specific, and there's no `HttpInput.Interceptor` present to process it, then there's zero hope in being able to parse the HttpServletRequest input stream.

Reply:
Dropwizard-based applications using their `BiDiGzipHandler` managed to decompress the input just fine.

Reply:
Opened #3030 to cover the eagerness of the check

Reply:
Regarding Dropwizard's `BiDiGzipHandler`, you might want to keep an eye on dropwizard/dropwizard#2479

Reply:
Fix PR ready for #3030

Reply:
Holy cow, you guys move fast. Thanks!